### ISSUE TITLE
Training input image resolution !!!

### ISSUE THREAD

**Issue by abhigoku10** [dbolya/yolact#140]
@dbolya Hi daniel, i am facing memory issue during training the custom dataset , please find my observations below .

1. I have trained the current source code multiple time with a custom dataset with varying resolutions less than 2k ,and it was working wonderfully 
2.Recently i have increased the dataset with 4K resoultions but now i am getting memory issues or dataloading issues , i tried couple things is by reducing the batch size and reducing the num_workers params but its not working 

Any thoughts on this

**Comment by dbolya** [comment]
Did you increase the network's resolution up from 550^2 or 700^2? Otherwise, you can just downscale all your images and annotations before feeding it into the network, as 4k images are useless if it's just going to be downscaled to <= 700x700.

**Comment by abhigoku10** [comment]
@dbolya nope i did not increase the ntetwork resolution ,but though the network is going to downscale the  4k resoultion to 700x700 memory error should not be happening since i am using 2080ti system

**Comment by dbolya** [comment]
Yeah but it might be a CPU memory (RAM) error not a GPU memory error. Also for evaluation, the image is scaled up to the size of the original image I think (which is done on the GPU), but that's with a batch size of 1 so probably not this issue.

Anyway, check your RAM usage.

**Comment by abhigoku10** [comment]
yes i checked up with the RAM usage the cPU ram usage is almost completely used , just wanted to know that operations like scaling and augmentation can be moved to GPU or turned off , if sowhich is the way to do it .. and my dataset is MSCOC+Custom data

**Comment by dbolya** [comment]
Well if you're using all your ram, I doubt moving it to the gpu would alleviate your issues since you probably have less gpu memory than cpu memory.

I recommend just decreasing your batch size or scaling down your images and annotations, since there really is no benefit from plugging in 4k images in the first place.

**Comment by abhigoku10** [comment]
@dbolya  there was one interesting thing which currently my training is progressing , i did the "num_workers=0" and the model is getting trained . what is the purpose of num_workers in ur code base and hwo to switch off the augmentations functionality as true or false

**Comment by dbolya** [comment]
Oh yeah good call. `num_workers` is how many extra CPU threads should be dedicated to computing the augmentations. Setting it to 0 will do the training and augmentation on the same thread, which I'd imagine uses less RAM. However, training will be slower as the data preparation will take longer.

If you want to turn off specific augmentations, look at the settings starting with `augment` in the config.

**Comment by abhigoku10** [comment]
@dbolya  hi i have a huge custom dataset by combination of both MsCOCO , PascalVOC and additional custom data too which is around 3 lakh images with 90 classes ,i always get **memory error** since the training final json is around 2gb can this be the issue i have kept batch size as 4 and numworkers=0

**Comment by dbolya** [comment]
Is this a GPU memory error or a CPU memory error (i.e., out of RAM)? I know that pycocotools loads the entire json into RAM at once, so that might be the issue if you don't have much CPU RAM.

**Comment by agneevguin** [comment]
I have image resolution of 1920x1208. I want to rescale/reshape the images during training for memory concerns.

- If I set max_size to 700 with batch 8, I get 

`CUDA out of memory. Tried to allocate ....`

- If I set max_size to 700, with batch 4, I get  

```
operation failed in interpreter:

def mul(a : float, b : Tensor) -> Tensor:
  return b * a
         ~~~~~ <--- HERE
def add(a : float, b : Tensor) -> Tensor:
  return b + a
def ne(a : float, b : Tensor) -> Tensor:
  return b != a
def eq(a : float, b : Tensor) -> Tensor:
  return b == a
def lt(a : float, b : Tensor) -> Tensor:
  return b > a
def le(a : float, b : Tensor) -> Tensor:
Some prob:  cuda runtime error (59) : device-side assert triggered at /opt/conda/conda-bld/pytorch_1556653215914/work/aten/src/THC/THCReduceAll.cuh:327
Some prob:  cuda runtime error (59) : device-side assert triggered at /opt/conda/conda-bld/pytorch_1556653215914/work/aten/src/THC/generic/THCTensorMath.cu:16
Some prob:  CUDA error: device-side assert triggered
cuda runtime error (59) : device-side assert triggered at /opt/conda/conda-bld/pytorch_1556653215914/work/aten/src/THC/generic/THCTensorMath.cu:16 
```

- If I set max_size to 550, with batch 4, I get  

`ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 1024000 and the array at index 2 has size 327680`

Is there something else that I need to modify to reshape the images?

**Comment by dbolya** [comment]
@agneevguin Those errors sound like they could be caused by using multiple GPUs and not having `batch_size / num_gpus` be a whole number. Can you make sure you have `export CUDA_VISIBLE_DEVICES=0` (or whatever GPU is best) first?
